packages/modules/devices/marstek/venus_c_e/bat.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+from typing import Optional, TypedDict, Any, Union
+from modules.common.abstract_device import AbstractBat
+from modules.common.component_state import BatState
+from modules.common.component_type import ComponentDescriptor
+from modules.common.fault_state import ComponentInfo, FaultState
+from modules.common.modbus import ModbusDataType, ModbusTcpClient_
+from modules.common.simcount import SimCounter
+from modules.common.store import get_bat_value_store
+from modules.devices.marstek.venus_c_e.config import VenusCEBatSetup
+
+
+class KwargsDict(TypedDict):
+    device_id: int
+    client: ModbusTcpClient_
+
+
+class VenusCEBat(AbstractBat):
+    def __init__(self, component_config: VenusCEBatSetup, **kwargs: Any) -> None:
+        self.component_config = component_config
+        self.kwargs: KwargsDict = kwargs
+
+    def initialize(self) -> None:
+        self.__device_id: int = self.kwargs['device_id']
+        self.client: ModbusTcpClient_ = self.kwargs['client']
+        self.sim_counter = SimCounter(self.__device_id, self.component_config.id, prefix="speicher")
+        self.store = get_bat_value_store(self.component_config.id)
+        self.fault_state = FaultState(ComponentInfo.from_component_config(self.component_config))
+
+    def _read_reg(self, addr: int, type_: ModbusDataType) -> Union[int, float]:
+        return self.client.read_holding_registers(addr, type_, unit=self.component_config.configuration.modbus_id)
+
+    def _write_reg(self, addr: int, val: int) -> None:
+        # Marstek Venus does not work with write_registers!
+        self.client._delegate.write_register(addr, val, unit=self.component_config.configuration.modbus_id)
+
+    def update(self) -> None:
+        power = -self._read_reg(32202, ModbusDataType.INT_32)
+        soc = self._read_reg(32104, ModbusDataType.UINT_16)
```

Comment:
Wie in der in PR #2683 beschriebenen Doku und Umsetzung braucht der SoC noch einen Faktor.
```suggestion
        soc = self._read_reg(32104, ModbusDataType.UINT_16) * 0.1
```

Reply:
Vermutlich ist die Doku hier inkorrekt. Zumindest bei meinem Venus-C darf nicht mit 0.1 multipliziert werden, sonst stimmt der angezeigte SoC in openWB nicht. Ich gehe davon aus, dass der Venus-E sich genauso verhält.